base/tools/src/main/java/com/netscape/cmstools/ocsp/OCSPCertVerifyCLI.java

```diff
@@ -0,0 +1,163 @@
+        option.setArgName("nickname");
+        options.addOption(option);
+
+        option = new Option(null, "path", true, "Path to OCSP responder (default: /ocsp/ee/ocsp");
+        option.setArgName("path");
+        options.addOption(option);
+
+        option = new Option(null, "request", true, "Path to DER-encoded OCSP request");
+        option.setArgName("path");
+        options.addOption(option);
+    }
+
+    @Override
+    public void execute(CommandLine cmd) throws Exception {
+
+        String[] cmdArgs = cmd.getArgs();
+
+        String caCertNickname = cmd.getOptionValue("ca-cert");
+        String ocspPath = cmd.getOptionValue("path", "/ocsp/ee/ocsp");
+        String requestPath = cmd.getOptionValue("request");
+
+        MainCLI mainCLI = (MainCLI) getRoot();
+        mainCLI.init();
+
+        ClientConfig config = getConfig();
+        String ocspURL = config.getServerURL() + ocspPath;
+
+        OCSPProcessor processor = new OCSPProcessor();
+        processor.setURL(ocspURL);
+
+        OCSPRequest request;
+
+        if (requestPath != null) {
+            logger.info("Loading OCSP request from " + requestPath);
+            byte[] data = Files.readAllBytes(Paths.get(requestPath));
+            request = processor.createRequest(data);
+
+        } else {
+            if (cmdArgs.length < 1) {
+                throw new Exception("Missing certificate serial number");
+            }
+
+            if (caCertNickname == null) {
+                throw new Exception("Missing CA signing certificate nickname");
+            }
+
+            CertId certID = new CertId(cmdArgs[0]);
+
+            logger.info("Creating OCSP request for cert " + certID.toHexString());
+            request = processor.createRequest(caCertNickname, certID.toBigInteger());
+        }
+
+        logger.info("Submitting OCSP request to " + ocspURL);
+        OCSPResponse response;
+        try {
+            response = processor.submitRequest(request);
+        } catch (Exception e) {
+            throw new CLIException("Unable to submit OCSP request: " + e.getMessage());
+        }
+
+        logger.info("Parsing OCSP response");
+        byte[] bytes = response.getResponseBytes().getResponse().toByteArray();
+        BasicOCSPResponse basic;
+
+        try (InputStream is = new ByteArrayInputStream(bytes)) {
+            basic = (BasicOCSPResponse) BasicOCSPResponse.getTemplate().decode(is);
+        }
+
+        ResponseData rd = basic.getResponseData();
+        int count = rd.getResponseCount();
+
+        for (int i = 0; i < count; i++) {
+            SingleResponse sr = rd.getResponseAt(i);
+
+            if (i > 0) {
+                System.out.println();
+            }
+
+            CertID certID = sr.getCertID();
+            INTEGER serialNumber = certID.getSerialNumber();
+            System.out.println("  Serial Number: " + new CertId(serialNumber).toHexString());
+
+            CertStatus status = sr.getCertStatus();
```

Comment:
Also, do we not already have an enums defined for the OCSP status strings ("Good", "Unknown", "Revoked") ? or at least constants?  We probably want to try to use them if they are already defined.

Reply:
I don't see any enumerations/constants that define those strings. We do have classes like `GoodInfo` but they don't define the strings either.